File: Gfg/Java/N meetings in one room.java
import java.io.*;
import java.util.*;
import java.lang.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());

        while (t-- > 0) {
            String inputLine[] = br.readLine().trim().split(" ");
            int n = Integer.parseInt(inputLine[0]);

            int start[] = new int[n];
            int end[] = new int[n];

            inputLine = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++)
                start[i] = Integer.parseInt(inputLine[i]);

            inputLine = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) 
                end[i] = Integer.parseInt(inputLine[i]);
                
            int ans = new Solution().maxMeetings(start, end, n);
            System.out.println(ans);
        }
    }
}
// } Driver Code Ends





class Solution 
{
    //Function to find the maximum number of meetings that can
    //be performed in a meeting room.
    public static int maxMeetings(int start[], int end[], int n)
    {
        int[][] arr = new int[n][2];
        
        for (int i = 0; i < n; i++) {
            arr[i][0] = start[i];
            arr[i][1] = end[i];
        }
        
        Arrays.sort(arr, (a,b) -> a[1] - b[1]);
        List<int[]> rooms = new ArrayList<>();
        rooms.add(arr[0]);
        int tl =  arr[0][1];
        for (int i = 1; i <n; i++) {
            if (arr[i][0] > tl) {
                rooms.add(arr[i]);
                tl = arr[i][1];
            }
        }
        return rooms.size();
    }
}

File: Gfg/C++/N meetings in one room.cpp
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends


class Solution{
    public:
        
        static bool comp(const pair<int, int>&a, const pair<int, int>&b){
            if(a.second == b.second)
                return a.first  <= b.first;
            return a.second < b.second;
        }
    
        int maxMeetings(int start[], int end[], int n) {
        // Your code here
        vector<pair<int, int>>vec;
        int i,c=0, fin;
        for(i=0;i<n;i++){
            vec.push_back({start[i], end[i]});
        }
        sort(vec.begin(), vec.end(), comp);
        
        for(i=0;i<n;){
            c++;
            fin=vec[i].second;
            i++;
            while(i < n && fin >= vec[i].first){
                i++;
            }
        }
        return c;
    }
};

// { Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int start[n], end[n];
        for (int i = 0; i < n; i++) cin >> start[i];

        for (int i = 0; i < n; i++) cin >> end[i];

        Solution ob;
        int ans = ob.maxMeetings(start, end, n);
        cout << ans << endl;
    }
    return 0;
}  // } Driver Code Ends
